AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/TransformerV2/DataStoreSchemaDriftTests.swift

```diff
@@ -0,0 +1,130 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import AmplifyPlugins
+import Combine
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStoreCategoryPlugin
+
+/*
+ # 16 Schema drift scenario
+
+ type SchemaDrift @model {
+   id: ID!
+   enumValue: EnumDrift
+ }
+
+ enum EnumDrift {
+    ONE
+    TWO
+    THREE
+ }
+
+ Codegenerated EnumDrift Enum has `THREE` enum case commented out to create an schema drift.
+ */
+
+@available(iOS 13.0, *)
+class DataStoreSchemaDriftTests: SyncEngineIntegrationV2TestBase {
+
+    var subscriptions: Set<AnyCancellable> = []
+
+    struct TestModelRegistration: AmplifyModelRegistration {
+        func registerModels(registry: ModelRegistry.Type) {
+            registry.register(modelType: SchemaDrift.self)
+        }
+
+        let version: String = "1"
+    }
+
+    func testSchemaDrift() throws {
+        setUp(withModels: TestModelRegistration())
+        let startSuccess = expectation(description: "start success")
+        try startAmplify {
+            startSuccess.fulfill()
+        }
+        wait(for: [startSuccess], timeout: TestCommonConstants.networkTimeout)
+        // Save some data with the missing enum case, do this by directly calling API
+        // with a custom variables object. Later, decoding will fail.
+        let saveSuccessWithTransformationError = expectation(description: "saved success with transformation error")
+        let schemaDrift = SchemaDrift(enumValue: .one)
```

Comment:
why are we not creating the model with enumvalue as `.three` directly here?

Reply:
it's because we're replicating a old client which hasn't upgraded to the latest app containing the new codegenerated files, so there is no `.three` case defined in the Enum, we manually edit the value to "THREE" and send it to AppSync directly